### PR TITLE
[Bug]fix pad wrong answer

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
sql case:
 SELECT LPAD('hi',1,'123')
 SELECT RPAD('hi',1,'123')

wrong answer:
hi
hi

expect answer:
h
h

### CODE REVIEW

File: be/src/vec/functions/function_string.h [hunk 1]
```diff
@@ -642,13 +642,13 @@ class FunctionStringPad : public IFunction {
                     }
 
                     // Append given string.
```
Comment (zbtzbtzbt) [inline be/src/vec/functions/function_string.h:644]: Here is my test result:
```
mysql> set enable_vectorized_engine = true;
Query OK, 0 rows affected (0.00 sec)

mysql> SELECT LPAD('hi',1,'123');
ERROR 1064 (HY000): errCode = 2, detailMessage = there is no scanNode Backend. [10002: not alive]
```
your commit looks like right, but after turn on vec switch, LPAD RPAD will cause BE node down.
I think pad algorithm is wrong in somewhere, you can try to find out and fix it totally.

File: be/src/vec/functions/function_string.h [hunk 2]
```diff
@@ -642,13 +642,13 @@ class FunctionStringPad : public IFunction {
...
-                    buffer.append(str_data, str_data + str_len);
+                    buffer.append(str_data, str_data + std::min(col_len_data[i],str_len));
                     StringOP::push_value_string(std::string_view(buffer.data(), buffer.size()), i,
                                                 res_chars, res_offsets);
```
Comment (zbtzbtzbt) [inline be/src/vec/functions/function_string.h:648]: you can try to test some chinese character.
I guess case `select RPAD('ab',3,'我');` can not pass. 
```
MySQL starting ...
RPAD('ab',3,'我')
ab我
```

File: be/src/vec/functions/function_string.h (hunk 1 above)
Comment (BiteTheDDDDt) [reply be/src/vec/functions/function_string.h:644]: 图片
Backend works well for me.
Can you confirm that the modification of the pad caused be to crash?

**Comment by BiteTheDDDDt** [comment]
@zbtzbtzbt  I fixed it, now pad is work good at chinese character.

```
MySQL [(none)]> set enable_vectorized_engine = true;
Query OK, 0 rows affected (0.000 sec)

MySQL [(none)]> SELECT RPAD('啊',1,'饿');
+-----------------------+
| rpad('啊', 1, '饿')   |
+-----------------------+
| 啊                    |
+-----------------------+
1 row in set (0.004 sec)

MySQL [(none)]> SELECT RPAD('a啊',1,'饿');
+------------------------+
| rpad('a啊', 1, '饿')   |
+------------------------+
| a                      |
+------------------------+
1 row in set (0.004 sec)

MySQL [(none)]> SELECT RPAD('啊a',1,'饿');
+------------------------+
| rpad('啊a', 1, '饿')   |
+------------------------+
| 啊                     |
+------------------------+
1 row in set (0.004 sec)

MySQL [(none)]> SELECT RPAD('啊a',2,'饿');
+------------------------+
| rpad('啊a', 2, '饿')   |
+------------------------+
| 啊a                    |
+------------------------+
1 row in set (0.005 sec)

MySQL [(none)]> SELECT RPAD('啊a',3,'饿');
+------------------------+
| rpad('啊a', 3, '饿')   |
+------------------------+
| 啊a饿                  |
+------------------------+
1 row in set (0.004 sec)
MySQL [(none)]> SELECT RPAD('啊a',3,'e饿');
+-------------------------+
| rpad('啊a', 3, 'e饿')   |
+-------------------------+
| 啊ae                    |
+-------------------------+
1 row in set (0.004 sec)

MySQL [(none)]> SELECT RPAD('啊a',4,'e饿');
+-------------------------+
| rpad('啊a', 4, 'e饿')   |
+-------------------------+
| 啊ae饿                  |
+-------------------------+
1 row in set (0.004 sec)
```

**Comment by BiteTheDDDDt** [comment]
I found that the original implementation still had errors in some corner cases, and I fixed it. 

图片

File: be/src/vec/functions/function_string.h [hunk 3]
```diff
@@ -609,6 +610,7 @@ class FunctionStringPad : public IFunction {
             pad_index.clear();
             buffer.clear();
             if (null_map_data[i] || col_len_data[i] < 0) {
+                null_map_data[i] = true;
```
Comment (HappenLee) [inline be/src/vec/functions/function_string.h:614]: 加个注释哈~~

**Pull request merged**